engine/access/rest/converter.go

```diff
@@ -3,21 +3,201 @@ package rest
 import (
 	"encoding/base64"
 	"fmt"
+	"regexp"
 
 	"github.com/onflow/flow-go/engine/access/rest/generated"
 	"github.com/onflow/flow-go/model/flow"
 )
 
-// Converter provides the conversion function to/from the Swagger object to Flow objects
+// Converter provides functionality to convert from request models generated using
+// open api spec and flow models.
 
-func toBlock(flowBlock *flow.Block) *generated.Block {
+// Flow section - converting request data to flow models with validation.
+
+func toID(id string) (flow.Identifier, error) {
+	valid, _ := regexp.MatchString(`^[0-9a-fA-F]{64}$`, id)
+	if !valid {
+		return flow.Identifier{}, fmt.Errorf("invalid ID")
+	}
+
+	return flow.HexStringToIdentifier(id)
+}
+
+func toAddress(address string) (flow.Address, error) {
+	valid, _ := regexp.MatchString(`^[0-9a-fA-F]{16}$`, address)
+	if !valid {
+		return flow.Address{}, fmt.Errorf("invalid address")
+	}
+
+	return flow.HexToAddress(address), nil
+}
+
+func toProposalKey(key *generated.ProposalKey) (flow.ProposalKey, error) {
+	address, err := toAddress(key.Address)
+	if err != nil {
+		return flow.ProposalKey{}, err
+	}
+
+	return flow.ProposalKey{
+		Address:        address,
+		KeyIndex:       uint64(key.KeyIndex),
+		SequenceNumber: uint64(key.SequenceNumber),
+	}, nil
+}
+
+func toSignature(signature string) ([]byte, error) {
+	signatureBytes := []byte(signature)
+	// the maximum signature length is arbitrarily chosen to be 2048. This check is to ensure that the signature is not
+	// absurdly long
+	if len(signatureBytes) > 2048 {
+		return nil, fmt.Errorf("invalid signature")
+	}
+	return signatureBytes, nil
+}
+
+func toTransactionSignature(transactionSignature *generated.TransactionSignature) (flow.TransactionSignature, error) {
+	address, err := toAddress(transactionSignature.Address)
+	if err != nil {
+		return flow.TransactionSignature{}, err
+	}
+
+	signature, err := toSignature(transactionSignature.Signature)
+	if err != nil {
+		return flow.TransactionSignature{}, err
+	}
+
+	return flow.TransactionSignature{
+		Address:     address,
+		SignerIndex: int(transactionSignature.SignerIndex),
+		KeyIndex:    uint64(transactionSignature.KeyIndex),
+		Signature:   signature,
+	}, nil
+}
+
+func toTransactionSignatures(sigs []generated.TransactionSignature) ([]flow.TransactionSignature, error) {
+	signatures := make([]flow.TransactionSignature, len(sigs))
+	for _, sig := range sigs {
+		signature, err := toTransactionSignature(&sig)
+		if err != nil {
+			return nil, err
+		}
+
+		signatures = append(signatures, signature)
+	}
+
+	return signatures, nil
+}
+
+func toTransaction(tx *generated.TransactionsBody) (flow.TransactionBody, error) {
+	args := make([][]byte, len(tx.Arguments))
+	for _, arg := range tx.Arguments {
+		// todo validate
+		args = append(args, []byte(arg))
```

Comment:
This is also not correct, since in the spec, arguments are base64 encoded

Reply:
can i tackle it as part of https://github.com/dapperlabs/flow-go/issues/6022?